scripts/process_gperf.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import argparse
+import os
+import re
+
+# --- debug stuff ---
+
+"""
+gperf C file post-processor
+
+We use gperf to build up a perfect hashtable of pointer values. The way gperf
+does this is to create a table 'wordlist' indexed by a string repreesentation
+of a pointer address, and then doing memcmp() on a string passed in for
+comparison
```

Comment:
This is super clever, but I just have to whine:

How robust is this vs. changes to the code output in upstream gperf?  I doubt they're doing much development these days, but if they do it's not unlikely that this script will break badly.  It might be a good idea to put a version check somewhere (maybe right here: is there a "generated by gperv v${WHATEVER}" comment in the file?) to catch that circumstance and force us to revalidate the metacompiler.

Reply:
good idea, I'll add a version string check and error out. so far this works in gperf 3.0.x and 3.1. agree this will fall over if the generated C code changes too much.